### PR TITLE
Make Recommendation option to the developer

### PR DESCRIPTION
**Pull request by hani-ibrahim**
fixing #751

### CODE REVIEW

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift [hunk 1]
```diff
@@ -111,10 +111,18 @@ struct GuestOrderPlacedLayout: CheckoutSummaryLayout {
...
     let allowsArticleRefine = false
 
     func submitButtonBackgroundColor(readyToCheckout: Bool) -> UIColor { return UIColor(hex: 0x509614) }
     func submitButtonTitle(isPaypal: Bool) -> String { return "summaryView.submitButton.backToShop" }
 
 }
+
+fileprivate extension AtlasUIViewController {
+
+    fileprivate static var displayRecommendations: Bool {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: Drop it and use `Config.shared`

**Comment by bonkey** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift [hunk 2]
```diff
@@ -111,18 +112,10 @@ struct GuestOrderPlacedLayout: CheckoutSummaryLayout {
     let showsDetailArrow = false
     let showsGuestStackView = true
     let showsOrderStackView = true
-    let showsRecommendationStackView = AtlasUIViewController.displayRecommendations
+    let showsRecommendationStackView = Config.shared?.displayRecommendations ?? true
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: Hm... Shouldn't it be `false` by default? Imagine a developer "forgets" to handle it – then user is puzzled.

However `true` should be a default value in `Options` I guess...

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: The developer forget to handle what? This `??` is just for the optional `config`. 
`displayRecommemdation` is `true` in options be default

File: AtlasSDK/AtlasSDK/Options.swift [hunk 3]
```diff
@@ -13,17 +13,20 @@ public struct Options {
...
                 useSandboxEnvironment: Bool? = nil,
                 interfaceLanguage: String? = nil,
                 configurationURL: URL? = nil,
+                displayRecommendations: Bool = true,
                 infoBundle bundle: Bundle = Bundle.main) {
         self.clientId = clientId ?? bundle.string(for: .clientId) ?? ""
         self.salesChannel = salesChannel ?? bundle.string(for: .salesChannel) ?? ""
         self.useSandboxEnvironment = useSandboxEnvironment ?? bundle.bool(for: .useSandboxEnvironment) ?? false
         self.interfaceLanguage = interfaceLanguage ?? bundle.string(for: .interfaceLanguage)
+        self.displayRecommendations = displayRecommendations
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/Options.swift]: Ah, it should be a Info.plist key for this as well.

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift (hunk 2 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: A developer forgets to handle opening PDP – then a user tapping on reco ends up without PDP.

I know `displayRecommendation` is `true`, but it is in an independent piece of code. In this context you just _assume_ it's `true`. If for any circumstances it changes: let say somebody makes it `false` in a fork, to simplify initialization of `Options` or in some cases we decide to make it false – then _assumed_ behavior is different.

And yes, I know it's only for optional config, so it is always correct.

Maybe `Options` should have default values exposed?

```swift

struct Options {
  
  struct Defaults {
    static let clientId = ""
    static let salesChannel = ""
    static let useSandboxEnvironment = false
    static let displayRecommendations = true
  }

  public init( // ...
        self.clientId = clientId ?? bundle.string(for: .clientId) ?? Defaults.clientId
        self.salesChannel = salesChannel ?? bundle.string(for: .salesChannel) ?? Defaults.salesChannel
        self.useSandboxEnvironment = useSandboxEnvironment ?? bundle.bool(for: .useSandboxEnvironment) ?? Defaults.useSandboxEnvironment
        self.displayRecommendations = useSandboxEnvironment ?? bundle.bool(for: .displayRecommendations) ?? Defaults.displayRecommendations
   // ...

```

And then:

```swift
     let showsRecommendationStackView = Config.shared?.displayRecommendations ?? Options.Defaults.displayRecommendations
```

Yes, I know, tiny and improbable issue eventually, but I'm just promoting safeness here – still without big overhead I would say.

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift (hunk 2 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: yeah I like the defaults approach

File: AtlasSDK/AtlasSDKTests/Info.plist [hunk 4]
```diff
@@ -28,5 +28,7 @@
 	<string>en</string>
 	<key>ATLASSDK_USE_SANDBOX</key>
 	<true/>
+	<key>ATLASSDK_DISPLAY_RECOMMENDATIONS</key>
```
Comment (bonkey) [inline AtlasSDK/AtlasSDKTests/Info.plist]: Don't forget to update [wiki](https://github.com/zalando-incubator/atlas-ios/wiki/Configuration#via-infoplist)

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

**Pull request merged**